### PR TITLE
feat: init swap protocol

### PR DESCRIPTION
**Pull request by alaibe**
* Add proto files
* Add options
* Add credit/debit for soft mode

### CODE REVIEW

File: waku/v2/protocol/swap/waku_swap.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/libp2p/go-libp2p-core/protocol"
+)
+
+const (
+	SoftMode int = 0
+	MockMode int = 1
+	HardMode int = 2
+)
+
+const WakuSwaoID_v200 = protocol.ID("/vac/waku/swap/2.0.0-beta1")
```
Comment (richard-ramos) [inline waku/v2/protocol/swap/waku_swap.go:17]: typo?

File: waku/v2/protocol/swap/waku_swap.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	}
+}
+
+func (s *WakuSwap) sendCheque(peerId string) {
+	log.Debug("not yet implemented")
+}
+
+func (s *WakuSwap) applyPolicy(peerId string) {
+	if s.Accounting[peerId] <= s.params.disconnectThreshold {
+		log.Warnf("Disconnect threshhold has been reached for %s at %d", peerId, s.Accounting[peerId])
```
Comment (richard-ramos) [inline waku/v2/protocol/swap/waku_swap.go:49]: Is this log correct? I'm confused due to the `<=` meaning that the threshold has been reached

File: waku/v2/protocol/swap/waku_swap.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	}
+}
+
+func (s *WakuSwap) Credit(peerId string, n int) {
+	s.accountingMutex.Lock()
+	defer s.accountingMutex.Unlock()
+
+	if _, ok := s.Accounting[peerId]; !ok {
+		s.Accounting[peerId] = 0
+	}
```
Comment (richard-ramos) [inline waku/v2/protocol/swap/waku_swap.go:66]: Not an error, but `s.Accounting[peerId]` will be 0 regardless if the `peerId` exists or not in the map

**Review by richard-ramos — APPROVED** [review]

File: waku/v2/protocol/swap/waku_swap.go (hunk 3 above)
Comment (alaibe) [reply waku/v2/protocol/swap/waku_swap.go:66]: :facepalm: alright I can just remove the check!

File: waku/v2/protocol/swap/waku_swap.go (hunk 2 above)
Comment (alaibe) [reply waku/v2/protocol/swap/waku_swap.go:49]: Yep it seems:
https://github.com/status-im/nim-waku/blob/master/waku/v2/protocol/waku_swap/waku_swap.nim#L252-L272

**Pull request merged**